libs/plugin-analysis-api/src/main/java/org/elasticsearch/plugin/analysis/api/AnalyzerFactory.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugin.analysis.api;
+
+import org.elasticsearch.plugin.api.Nameable;
+
+/**
+ * An analysis component used to create Analyzers.
+ */
+public interface AnalyzerFactory<T extends org.apache.lucene.analysis.Analyzer> extends Nameable {
```

Comment:
Do we need the generic parameter here? The other factories are just returning a base type, which I think should be enough?

Reply:
agree, I don't think we need a generic here